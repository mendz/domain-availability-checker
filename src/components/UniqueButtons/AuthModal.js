import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import {
  showAuthModal,
  closeAuthModal,
  authLogOut,
} from '../../store/actions/auth.js';

import Button from '../UI/Button/Button';
import Modal from '../UI/Modal/Modal';
import Auth from '../../containers/Auth/Auth';

const AuthModal = ({
  authModal,
  isLoggedIn,
  showAuthModal,
  closeAuthModal,
  authLogOut,
  disabled,
}) => {
  const style = {
    minWidth: '165px',
  };

  let content = (
    <>
      <Button onClick={showAuthModal} style={style} disabled={disabled}>
        Log In / Sign Up
      </Button>
      <Modal show={authModal} closed={closeAuthModal}>
        <Auth />
      </Modal>
    </>
  );

  if (isLoggedIn) {
    content = (
      <Button onClick={authLogOut} style={style} disabled={disabled}>
        Log Out
      </Button>
    );
  }

  return content;
};

const mapStateToProps = state => ({
  authModal: state.auth.authModal,
  isLoggedIn: state.auth.userId ? true : false,
});

const mapDispatchToProps = dispatch =>
  bindActionCreators({ showAuthModal, closeAuthModal, authLogOut }, dispatch);

AuthModal.propTypes = {
  disabled: PropTypes.bool,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AuthModal);

export { AuthModal };
