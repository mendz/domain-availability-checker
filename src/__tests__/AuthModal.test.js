import React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';

import { AuthModal } from '../components/UniqueButtons/AuthModal';

describe('<AuthModal />', () => {
  let wrapper;

  const initialProps = {
    isLoggedIn: false,
    authModal: true,
    authLogOut: jest.fn(() => {}),
    closeAuthModal: jest.fn(() => {}),
  };

  beforeEach(() => {
    wrapper = shallow(<AuthModal {...initialProps} />);
  });

  it('matches the snapshot', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should display the Button with the text: "Log In / Sign Up"', () => {
    const html = wrapper.find('Button').html();
    expect(html).toContain('Log In / Sign Up');
  });

  it('should display the Button with the text: "Log Out" when props.isLoggedIn is true', () => {
    wrapper.setProps({ isLoggedIn: true });
    const html = wrapper.find('Button').html();
    expect(html).toContain('Log Out');
  });

  it('the button should be disabled if props.disabled="true"', () => {
    wrapper.setProps({ disabled: true });
    expect(wrapper.find('Button').props().disabled).toBeTruthy();
  });
});
