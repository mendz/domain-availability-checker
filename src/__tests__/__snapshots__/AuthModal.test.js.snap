// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthModal /> matches the snapshot 1`] = `
<Fragment>
  <Button
    style={
      Object {
        "minWidth": "165px",
      }
    }
  >
    Log In / Sign Up
  </Button>
  <Modal
    closed={[MockFunction]}
    show={true}
  >
    <Connect(Auth) />
  </Modal>
</Fragment>
`;
